ddtrace/tracer/spancontext.go

```diff
@@ -234,6 +234,9 @@ func (t *trace) finishedOne() {
 		if t.priority != nil {
 			t.root.Metrics[keySamplingPriority] = *t.priority
 		}
+		if tr.hostname != "" {
```

Comment:
I guess it's fine here, for now, because users won't be able to override it. If we want them to do so later we should move it to `StartSpan`.

Reply:
Yup! Right now, we don't protect against users setting tags that are effectively "internal" and "special".
So for now, if a user did set that tag themselves, it'd get overwritten with the correct value.

Reply:
Thanks for confirming.